File: 二叉树/108.将有序数组转换为二叉搜索树.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return dfs(nums,0,nums.length);
    }

    TreeNode dfs(int[] nums,int left ,int right){
        if(left== right){
            return null;
        }
        int m=(left+right)>>>1;
        return new TreeNode(nums[m],dfs(nums,left,m),dfs(nums,m+1,right));
    }
}

File: 二叉树/108.将有序数组转换为二叉搜索树.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
package main

func sortedArrayToBST(nums []int) *TreeNode {
	return dfs(nums, 0, len(nums))
}

func dfs(num []int, left int, right int) *TreeNode {
	if left == right {
		return nil
	}
	m := (left + right) / 2
	return &TreeNode{num[m], dfs(num, left, m), dfs(num, m+1, right)}
}
